tts/src/main/java/com/caneseeproject/tts/tts.kt

```diff
@@ -1,74 +1,49 @@
+                is Vision.ObjectDetection -> speakOut("object detection is turned on")
+                is Vision.Scenery -> speakOut("scene Description")
+                is ODReading.ObstacleDistance -> speakOut("Distance Detection is turned on")
 
-//              Glasses
-            Device.Glasses -> when (recent) {
-                is NotificationType.SensorNotification -> when (glassesMode) {
-                    is Vision.OCR -> speakOut("OCR is activated")
-                    is Vision.Facial -> speakOut("face recognition is turned on")
-                    is Vision.Emotion -> speakOut("Emotion recognition is turned on")
-                    is Vision.ObjectDetection -> speakOut("Object detection is turned on")
-                    is Vision.Scenery -> speakOut("Scene Description is turned on ")
-                }
-                is NotificationType.SensorContent -> when (glassesMode) {
-                    is Vision.OCR -> speakOut(glassesMode.transcript)
-                    is Vision.Facial -> speakOut(glassesMode.prettyFace)
-                    is Vision.Emotion -> speakOut(glassesMode.emotion)
-                    is Vision.ObjectDetection -> speakOut(glassesMode.objects.toString())
-                    is Vision.Scenery -> speakOut(glassesMode.scene)
-                }
-                is NotificationType.AppMessage -> speakOut("glasses connection is established")
             }
+            is NotificationType.SensorContent -> when (recent.textReading) {
+                is Vision.OCR -> speakOut(recent.textReading.transcript)
+                is Vision.Facial -> speakOut(recent.textReading.prettyFace)
+                is Vision.Emotion -> speakOut(recent.textReading.emotion)
+                is Vision.ObjectDetection -> speakOut(recent.textReading.objects.toString())
+                is Vision.Scenery -> speakOut(recent.textReading.scene)
+                is ODReading.ObstacleDistance -> speakOut(recent.textReading.distance.toString())
+            }
+            is NotificationType.AppMessage -> when{
 
-//              Cane
-            Device.Cane -> when (recent) {
-                is NotificationType.SensorNotification -> when (obstacleMode) {
-                    is ODReading.ObstacleDistance -> speakOut("Cane Distance is turned on")
-                }
-                is NotificationType.SensorContent -> when (obstacleMode) {
-                    is ODReading.ObstacleDistance -> speakOut(obstacleMode.distance.toString())
-
-                }
-                is NotificationType.AppMessage -> speakOut("Cane connection is established")
             }
         }
     }
 }
 sealed class NotificationType{
-
     class AppMessage :NotificationType()
-    class SensorNotification(val textInput :SensorInput) :NotificationType()
+    class SensorNotification(val sensorType :SensorReading) :NotificationType()
     class SensorContent(val textReading :SensorReading) :NotificationType()
```

Comment:
Both `SensorNotification` and `SensorContent` take a parameter of `SensorReading`!
One should accept a `SensorInput`.